Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1203 @@
+        /// </summary>
+        public static string NetworkLocationSignature { get; internal set; } = Guid.NewGuid().ToString();
+
+        /// <summary>
+        /// Gets a value indicating whether IP6 is enabled.
+        /// </summary>
+        public static bool IsIP6Enabled { get; internal set; }
+
+        /// <summary>
+        /// Gets a value indicating whether IP4 is enabled.
+        /// </summary>
+        public static bool IsIP4Enabled { get; internal set; } = true;
+
+        /// <summary>
+        /// Gets a value indicating whether is multi-socket binding available.
+        /// </summary>
+        public static bool EnableMultiSocketBinding { get; internal set; } = true;
+
+        /// <summary>
+        /// Gets the number of times the network address has changed.
+        /// </summary>
+        public static int NetworkChangeCount { get; internal set; } = 1;
+
+        /// <inheritdoc/>
+        public NetCollection RemoteAddressFilter { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether is all IPv6 interfaces are trusted as internal.
+        /// </summary>
+        public bool TrustAllIP6Interfaces { get; internal set; }
+
+        /// <summary>
+        /// Gets the Published server override list.
+        /// </summary>
+        public Dictionary<IPNetAddress, string> PublishedServerOverrides => _overrideUrls;
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        public List<PhysicalAddress> GetMacAddresses()
+        {
+            // Populated in construction - so always has values.
+            lock (_intLock)
+            {
+                return _macAddresses.ToList();
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool IsGatewayInterface(object? addressObj)
+        {
+            var address = (addressObj is IPAddress addressIP) ?
+                addressIP : (addressObj is IPObject addressIPObj) ?
+                    addressIPObj.Address : IPAddress.None;
+
+            lock (_intLock)
+            {
+                return _internalInterfaces.Where(i => i.Address.Equals(address) && (i.Tag < 0)).Any();
+            }
+        }
+
+        /// <inheritdoc/>
+        public NetCollection GetLoopbacks()
+        {
+            NetCollection nc = new NetCollection();
+            if (IsIP4Enabled)
+            {
+                nc.Add(IPAddress.Loopback);
+            }
+
+            if (IsIP6Enabled)
+            {
+                nc.Add(IPAddress.IPv6Loopback);
+            }
+
+            return nc;
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcluded(IPAddress ip)
+        {
+            return _excludedSubnets.Contains(ip);
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcluded(EndPoint ip)
+        {
+            if (ip != null)
```

Comment:
`ip` isn't nullable here

Reply:
Intellisense gives me "In externally visible method 'bool NetworkManager.IsExcluded(EndPoint ip)', validate parameter 'ip' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument." when i remove it because of _excludedSubnets.Contains(((IPEndPoint)ip).Address);

Reply:
I think these messages are appearing because the entire solution isn't nullable and they're public methods. (seen others elsewhere the same)

How do you want to play it? Keep the code as is without messages, or accept without checks